api/src/main/java/marquez/db/DatasetVersionDao.java

```diff
@@ -103,7 +102,7 @@ default DatasetVersionRow upsertDatasetVersion(
               UUID.randomUUID(),
               now,
               field.getName().getValue(),
-              field.getType().name(),
+              Utils.toUpperCaseIfPresent(field.getType()),
```

Comment:
proposal, maybe we can add this Utils.toUpperCaseIfPresent() logic to setter? Cause in all places we should do that even on getters, but if we apply that on object creation, we will be sure that it's uppercased